src/pages/index.js

```diff
@@ -0,0 +1,329 @@
+    const cardLikeButton = cardElement.querySelector(".card__like-button");
+    const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+    if (data.isLiked) {
+      cardLikeButton.classList.add("card__like-button_liked");
+    }
+  
+    cardImageElement.setAttribute("src", data.link);
+    cardImageElement.setAttribute("alt", data.name);
+    cardNameElement.textContent = data.name;
+    
+    cardImageElement.addEventListener("click", () => {
+        openModal(previewModal);
+
+        previewModalImageElement.src = data.link;
+        previewModalImageElement.alt = data.name;
+        previewModalCaptionElement.textContent = data.name;
+    });
+
+    cardLikeButton.addEventListener("click", (evt) => handleLikeCard(evt, data._id, data.isLiked));
+
+    cardDeleteButton.addEventListener("click", () => handleDeleteCard(cardElement, data._id));
+
+    return cardElement;
+
+}
+
+
+function openModal (modal) {
+    modal.classList.add("modal_opened");
+    document.addEventListener("keydown", handleEscape);
+}
+
+function closeModal (modal) {
+    modal.classList.remove("modal_opened");
+    document.removeEventListener("keydown", handleEscape);
+}
+
+function handleLikeCard(event, cardId, isLiked) {
+  api.updateLikeCard(cardId, isLiked)
+    .then(() => {
+      event.target.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+
+function handleProfileFormSubmit(evt) {
+    evt.preventDefault();
+
+    const submitButton = evt.submitter;
+    setButtonText(submitButton, true, "Saving...");
+
+    api.updateUserInfo({name: editProfileNameInput.value, about: editProfileDescriptionElement.value})
+    .then((userData) => {
+        profileNameElement.textContent = userData.name;
+        profileDescriptionElement.textContent = userData.about;
+        closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, true, "Saving...", "Save");
+    });
+
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const cardSubmitButton = evt.submitter;
+  cardSubmitButton.disabled = true;
+  setButtonText(cardSubmitButton, "Saving...");
+
+  const inputValue = { 
+      name: cardCaptionInput.value, 
+      link: cardLinkInput.value
+  };
+
+  api.createCard(inputValue)
+      .then((newCard) => {
+          const cardEl = getCardElement(newCard);
+          cardList.prepend(cardEl);
+          closeModal(cardModal);
+          cardFormElement.reset();
+      })
+      .catch(console.error)
+      .finally(() => {
+          cardSubmitButton.disabled = false;
+          setButtonText(cardSubmitButton, "Save");
+      });
+}
+
+document.getElementById("edit-avatar-form").addEventListener("submit", handleAvatarFormSubmit);
+
+
+function handleAvatarFormSubmit(evt) {
+    evt.preventDefault();
+
+
+    const avatarLinkInput = avatarModal.querySelector("#profile-avatar-input");
```

Comment:
It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine